### ISSUE TITLE
Accessibility check?

### ISSUE THREAD

**Issue by rycoley** [datascijedi/website#29]
Do we need to do any additional checks to ensure that the website content is screen-reader friendly and accessible in any other ways? Mine mentioned that most of the defaults for Quarto supported accessibility. Is there anything else we need to check?

**Comment by mine-cetinkaya-rundel** [comment]
@rycoley I ran an accessibility check using axe DevTools and addressed the ones I could with #37.

There are two alt-text issues that are due to how Quarto works, I'll open issues for those for Quarto enhancements.

There are also some remaining color contrast issues in the Twitter embed, I don't think we can address them but maybe someone else knows?

Finally, there are some issues remaining that are around making links accessible to screenreaders with aria labels etc. which I haven't been able to address fully but if there's someone in the group more well-versed in those, maybe we can figure them out together. Otherwise, I'm happy to give it another shot in the next few days.

**Comment by fatso-jetson** [comment]
@mine-cetinkaya-rundel @rycoley I went through the links on the site pages and added aria labels in places where I thought that the purpose of the link was unclear (using [this helpful overview]( https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA8#:~:text=The%20aria%2Dlabel%20attribute%20provides,used%20instead%20of%20aria%2Dlabel%20)). For example, for the upcoming webinar I added "Register for [webinar title]" as an aria label to the "Register now" button. All changes made are in 1e22a99.

I was unsure whether we also needed to add aria labels to links like the following (in bold):

> The Communications Committee manages the JEDI Outreach Group social media channels and website, coordinates the [**JEDI Corner**](http://localhost:4352/jedi-corner.html) in Amstat News, and advertises JEDI-related events and opportunities.

Do we need to label this as, for example, "Link to JEDI Corner listings page". Or is the "JEDI Corner" text associated with the link sufficient?

**Comment by rycoley** [comment]
Thanks for the aria labels instructions, Brian. I will be sure to do this when adding future links and will include it in the instructions for contributors. 
Based on the link you sent, I think we could go either way with the JEDI Corner link. I don't think there is any harm in having a more explicit label like "Link to JEDI Corner archive" and we should lean to being more clear when we can.

**Comment by ravimakhija** [comment]
> There are two alt-text issues that are due to how Quarto works, I'll open issues for those for Quarto enhancements.

I don't know enough about how accessibility features work as of yet (it's on the reading list!) to chime in on the open questions here, but @mine-cetinkaya-rundel I think it'd be helpful to link to the relevant Quarto GitHub issues from here as well.

**Comment by ravimakhija** [comment]
Hi All - if this issue is still open, I'm planning to take a closer look at any remaining website accessibility needs.

**Comment by mine-cetinkaya-rundel** [comment]
@ravimakhija I have not had a chance to run through the accessibility checks again on this site recently so there may still be remaining issues. I would recommend rebuilding with latest Quarto before checking.

**Comment by ravimakhija** [comment]
> I would recommend rebuilding with latest Quarto before checking.

Sounds good! The Netlify deploys are using [Quarto v1.2.475](https://quarto.org/docs/download/release.html) which is the currently the latest release, so I can give the accessibility checks another run soon.

**Comment by mine-cetinkaya-rundel** [comment]
The latest release is now 1.3+ and I'd recommend using that one since there were many updates to Quarto in between, including some accessibility ones.

**Comment by ravimakhija** [comment]
Sounds good. According to the [quarto-cli GitHub page](https://github.com/quarto-dev/quarto-cli/releases) the latest release states 1.2.475. However, the latest pre-release is 1.3.296. I can update Netlify to use 1.3.296 before doing the accessibility check. Thanks for the feedback!

**Comment by ravicodelabs** [comment]
Hi all, to follow up on this: I used Firefox accessibility inspector and some additional references in my checks, and I've made a few updates in PR #110, as well as provided some feedback to the Quarto team for improving accessibility. 

I did my testing using the live website, as well as locally using the latest pre-release Quarto version at the time (1.4.388). 

> Do we need to do any additional checks to ensure that the website content is screen-reader friendly and accessible in any other ways?

Between Mine's check with Axe tools and my check with Firefox accessibility tools, and additional checks by the team, my opinion (as someone who is not an expert but now has a decent understanding of web accessibility) is that we are in a pretty good place as far as web accessibility goes. 

> There are also some remaining color contrast issues in the Twitter embed, I don't think we can address them but maybe someone else knows?

We recently disabled the embedded timeline (see issue #100), so I suppose this is no longer an issue, unless we decide to bring it back at some point. 

> Finally, there are some issues remaining that are around making links accessible to screenreaders with aria labels etc. which I haven't been able to address fully but if there's someone in the group more well-versed in those, maybe we can figure them out together.

I think the use of `aria-label` for links (when needed) and `alt` for images (always) addresses this. I have added an "Accessibility Tips" section to [`CONTRIBUTING.md`](https://github.com/datascijedi/website/blob/main/CONTRIBUTING.md) for reference.
